vowpalwabbit/learner.h

```diff
@@ -465,9 +515,51 @@ VW_WARNING_STATE_POP
     ret.learn_fd.multipredict_f = nullptr;
     ret.pred_type = pred_type;
     ret.is_multiline = std::is_same<multi_ex, E>::value;
+    ret._hash_index = std::type_index(typeid(T));
```

Comment:
So this means the hash of a learner is defined by the contained data object type?

Reply:
This was mostly for expediency. I started with an enum class in the beginning, but realized it would require changing every reduction. Since no reductions currently share the same types (with the exception of the noop ones I created), this works as a temporary solution